planner/core/rule_eliminate_projection.go

```diff
@@ -133,12 +155,43 @@ func eliminatePhysicalProjection(p PhysicalPlan) PhysicalPlan {
 		oldCol.VirtualExpr = newCols[i].VirtualExpr
 		newRoot.Schema().Columns[i] = oldCol
 	}
-	return newRoot
+	return newRoot, nil
 }
 
 type projectionEliminator struct {
 }
 
+// ReplaceColumnOfExprInPhysicalProjection replaces column of expression by another PhysicalProjection.
+func ReplaceColumnOfExprInPhysicalProjection(expr expression.Expression, proj *PhysicalProjection, schema *expression.Schema) (expression.Expression, error) {
```

Comment:
Can `ReplaceColumnOfExprInPhysicalProjection` be private?

Reply:
I agree with you.

Reply:
Thanks for your updates! I am looking for reviewer helping on this PR.